pkg/appruntime/chain/common.go

```diff
@@ -168,3 +168,28 @@ func runTools(ctx context.Context, args map[string]any, tools []agent.Tool) map[
 	}
 	return args
 }
+
+/*
+When using **stream** mode and an error occurs, **fastchat** returns http code 200 and returns an error message in json,
+but this json is inconsistent with the default format of openai, so it is silently **ignored** by langchaingo,
+so in this case, we need to re-request the blocking mode to find the correct error message.
+
+fastchat will return like this:
+
+data: {"id": "chatcmpl-3Lo28HviQo8949WkJyfFzJ", "model": "7597c2a3-b186-4bac-b2e4-21ff56413f7f", "choices": [{"index": 0, "delta": {"role": "assistant"}, "finish_reason": null}]}
+data: {"text": "**NETWORK ERROR DUE TO HIGH TRAFFIC. PLEASE REGENERATE OR REFRESH THIS PAGE.**\n\n(\"addmm_impl_cpu_\" not implemented for 'Half')", "error_code": 50001}
+data: [DONE]
+*/
+func HandleNoErrNoOut(ctx context.Context, needStream bool, oldOut string, oldErr error, chain chains.Chain, args map[string]any, options []chains.ChainCallOption) (out string, err error) {
+	if !needStream {
+		return oldOut, oldErr
+	}
+	if needStream && len(strings.TrimSpace(oldOut)) == 0 && oldErr == nil {
+		if len(options) == 1 || len(options) == 0 {
```

Comment:
why we will not use options if the length = 1?

Reply:
Only the stream mode will encounter this problem. When in stream mode and the option length is 1, the option is the configuration parameter of the stream mode, and we can directly ignore it.

Reply:
let's use len(options) <=1 and add comments above this statement